sdk-lib/src/main/java/cash/z/ecc/android/sdk/block/CompactBlockProcessor.kt

```diff
@@ -278,19 +299,36 @@ class CompactBlockProcessor(
             ProcessorInfo(
                 networkBlockHeight = downloader.getLatestBlockHeight(),
                 lastScannedHeight = getLastScannedHeight(),
-                lastDownloadedHeight = max(getLastDownloadedHeight(), lowerBoundHeight - 1)
+                lastDownloadedHeight = BlockHeight.new(
+                    network,
+                    max(
+                        getLastDownloadedHeight().value,
+                        lowerBoundHeight.value - 1
+                    )
+                ),
+                lastDownloadRange = null,
+                lastScanRange = null
             ).let { initialInfo ->
                 updateProgress(
                     networkBlockHeight = initialInfo.networkBlockHeight,
                     lastScannedHeight = initialInfo.lastScannedHeight,
                     lastDownloadedHeight = initialInfo.lastDownloadedHeight,
-                    lastScanRange = (initialInfo.lastScannedHeight + 1)..initialInfo.networkBlockHeight,
-                    lastDownloadRange = (
-                        max(
-                            initialInfo.lastDownloadedHeight,
-                            initialInfo.lastScannedHeight
-                        ) + 1
+                    lastScanRange = if (initialInfo.lastScannedHeight != null && initialInfo.networkBlockHeight != null) {
```

Comment:
I wonder if these conditions are part of logic we might want to "document" in code by naming them under a computed variable or helper function.

Reply:
I think it only gets called once.  Do you have a recommendation for a nicer name?